### PR TITLE
GH-676: KafkaEmbeddedHolder - creating and adding topics.

### PR DESCRIPTION
**Pull request by nklmish**
Fixes https://github.com/spring-projects/spring-kafka/issues/676. we have also migrated `topics` from `array` to `set`.

### CODE REVIEW

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java [hunk 1]
```diff
@@ -235,31 +238,43 @@ public void before() throws Exception { //NOSONAR
...
 	}
 
+
+	/**
+	 * Add topics to the existing broker(s) using the configured number of partitions.
+	 * @param topics the topics.
+	 * @since 2.1
+	 */
+	public void addTopics(String... topics) {
+		createKafkaTopics(new HashSet<>(Arrays.asList(topics)), this.topics::addAll);
```
Comment (artembilan) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java]: I don't think we need such a `callback` overhead. You simply can call this `addAll()` in the end of this `addTopics()` method. More over in other place the callback is just empty, so, it really doesn't bring too much value in the logic of this class.

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java [hunk 2]
```diff
@@ -477,9 +496,7 @@ public void consumeFromAnEmbeddedTopic(Consumer<?, ?> consumer, String topic) th
 	 * @throws Exception an exception.
 	 */
 	public void consumeFromEmbeddedTopics(Consumer<?, ?> consumer, String... topics) throws Exception {
-		for (String topic : topics) {
-			assertThat(this.topics).as("topic '" + topic + "' is not in embedded topic list").contains(topic);
-		}
+		assertThat(this.topics).as("topic(s):'" + Arrays.toString(topics) + "' are not in embedded topic list").containsAll(new HashSet<>(Arrays.asList(topics)));
```
Comment (artembilan) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java]: Well. I would prefer to assert all, but report only wrong topics - don't tell that the whole set is wrong.

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java [hunk 3]
```diff
@@ -457,7 +475,7 @@ public boolean isEmbedded() {
 	 * @throws Exception an exception.
 	 */
 	public void consumeFromAllEmbeddedTopics(Consumer<?, ?> consumer) throws Exception {
-		consumeFromEmbeddedTopics(consumer, this.topics);
+		consumeFromEmbeddedTopics(consumer, this.topics.toArray(new String[this.topics.size()]));
```
Comment (artembilan) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java]: My latest IDEA doesn't recommend me to use arrays with static size for this operation:
image

Comment (nklmish) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java]: Thank you, updated my intellij and saw the same warning. Should be better now.

**Comment by garyrussell** [comment]
No concerns; LGTM; @nklmish thanks for the contribution.

**Comment by artembilan** [comment]
Sure thing, @nklmish !

Thank you very much for the contribution.
Looking forward for more!

**Comment by nklmish** [comment]
@garyrussell @artembilan Thank you very much

**Comment by artembilan** [comment]
... and cherry-picked to `2.1.x`.

**Pull request merged**